azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/HeartBeatProvider/SpringBootHeartBeatProvider.java

```diff
@@ -0,0 +1,123 @@
+package com.microsoft.applicationinsights.boot.HeartBeatProvider;
+
+import com.microsoft.applicationinsights.internal.heartbeat.HeartBeatPayloadProviderInterface;
+import com.microsoft.applicationinsights.internal.heartbeat.HeartBeatProviderInterface;
+import com.microsoft.applicationinsights.internal.heartbeat.MiscUtils;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.springframework.boot.SpringBootVersion;
+import org.springframework.core.SpringVersion;
+import org.springframework.core.env.Environment;
+
+/**
+ * <h1>SpringBoot Heartbeat Property Provider</h1>
+ * <p>
+ *   This class is a concrete implementation of {@link HeartBeatPayloadProviderInterface}
+ *   It enables setting SpringBoot Metadata to heartbeat payload.
+ * </p>
+ *
+ * @author Dhaval Doshi
+ */
+public class SpringBootHeartBeatProvider implements HeartBeatPayloadProviderInterface {
```

Comment:
Is it possible to extend the implementation in core? or make it a delegate? You shouldn't need to reimplement the whole class.

Reply:
Well I think know. The design for this is such that,  each separate functionality has it's own HB provider. For example springboot has it's own, Azure VM will have it's own and WebApps have it's own. This provides the flexibility for user to turn off the whole provider in itself incase if they do not need it rather then handpicking each properties to disable. The contract each Provider has to follow is to implement the interface. I think this provides nice abstraction, separates the module in itself from core and lets us be able to freely modify it without affecting others.

Reply:
Ok, i see. This is sending spring boot specific data; a decorator of heartbeat events. Makes sense.